gtda/mapper/cover.py

```diff
@@ -374,9 +378,11 @@ class CubicalCover(BaseEstimator, TransformerMixin):
 
     """
 
-    _hyperparameters = {'kind': [str, ['uniform', 'balanced']],
-                        'n_intervals': [int, (1, np.inf)],
-                        'overlap_frac': [float, (0, 1)]}
+    _hyperparameters = {
+        'kind': {'type': str, 'in': ['uniform', 'balanced']},
+        'n_intervals': {'type': int, 'in': Interval(1, np.inf, closed='left')},
+        'overlap_frac': {'type': float, 'in': Interval(0, 1, closed='neither')}
+    }
```

Comment:
Could it  be possible to reuse `_hyperparameters` from `OneDCover`?

Reply:
This would be fine now, but maybe not future proof if we decide there are smarter ways to do hyperparameter validation differently for `CubicalCover` for whatever reason?

Reply:
This is not a comment ihave strong feelings for. But if you know that some parameters. will *only* be used by another class (which has hyper_params defined), I think we should use that definition. When (if) we introduce the possibility to have different 'ondedimensional' pieces that constitute the cover, we will need to rethink it, you are right. But I do not expect to see anything other than `OneDCove` as 'building blocks' for this cover (mathematically speaking).